base_import_csv_optional/models/models.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# License and authorship info in:
+# __openerp__.py file at the root folder of this module.
+
+from openerp import api
+from openerp.models import BaseModel
+import logging
+
+_logger = logging.getLogger(__name__)
+base_load = BaseModel.load
+
+
+@api.model
+def load_import_optional(self, fields=None, data=None):
+    '''Overriding this method we only allow its execution
+    if current user belongs to the group allowed for CSV data import.
+    An exception is raised otherwise, and also log the import attempt.
+    '''
+    res = {}
+    current_user = self.env['res.users'].browse(self.env.uid)
+    allowed_group = 'base_import_csv_optional.group_import_csv'
+    if current_user and current_user.has_group(allowed_group):
```

Comment:
It would be safer to check for the existence of the group referenced with the external ID _base_import_csv_optional.group_import_csv_ before to use it. I suggest that when this group is not found (and therefore no user can reasonably belong to it) this method should just return the parent logic, which means to allow any user to run the import.